library/cohesity_oracle.py

```diff
@@ -0,0 +1,415 @@
+                for node in source['nodes']:
+                    if node['protectionSource']['name'] == self['endpoint']:
+                        return node['protectionSource']['id']
+
+        return False
+    except urllib_error.URLError as e:
+        # => Capture and report any error messages.
+        raise__cohesity_exception__handler(e.read(), module)
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+# => Register the new Endpoint as a Cohesity Protection Source.
+def register_source(module, self):
+    server = module.params.get('cluster')
+    validate_certs = module.params.get('validate_certs')
+    token = self['token']
+    try:
+        uri = "https://" + server + "/irisservices/api/v1/backupsources"
+        headers = {"Accept": "application/json",
+                   "Authorization": "Bearer " + token}
+        #payload = self.copy()
+        #payload['environment'] = "k" + self['environment']
+        #payload['hostType'] = "k" + self['hostType']
+        #payload['physicalType'] = "k" + self['physicalType']
+        endpoint = module.params.get('endpoint')
+        force_register = module.params.get('force_register')
+
+        # Payload template to register a physical source.
+        payload = {
+            "entity": {
+                "type": 6,
+                "physicalEntity": {
+                    "name": "",
+                    "type": 1,
+                    "hostType": 1
+                }
+            },
+            "entityInfo": {
+                "endpoint": "",
+                "type": 6,
+                "hostType": 1
+            },
+            "sourceSideDedupEnabled": True,
+            "throttlingPolicy": {
+                "isThrottlingEnabled": False
+            },
+        }
+        payload["forceRegister"] = force_register
+        payload["entity"]["physicalEntity"]["name"] = endpoint
+        payload["entityInfo"]["endpoint"] = endpoint
+
+        data = json.dumps(payload)
+        response = open_url(url=uri, data=data, headers=headers,
+                            validate_certs=validate_certs, timeout=REQUEST_TIMEOUT)
+
+        response = json.loads(response.read())
+
+        # => This switcher will allow us to return a standardized output
+        # => for all Protection Sources.
+        # response = dict(ProtectionSource=response[
+        #                     'physicalProtectionSource'])
+        return dict(response)
+    except urllib_error.URLError as e:
+        # => Capture and report any error messages.
+        raise__cohesity_exception__handler(e.read(), module)
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+# => Unregister an existing Cohesity Protection Source.
+def unregister_source(module, self):
+    server = module.params.get('cluster')
+    validate_certs = module.params.get('validate_certs')
+    token = self['token']
+    try:
+        uri = "https://" + server + \
```

Comment:
Use python sdk as much as possible for public endpoints.

Reply:
Okay Ashish, will change